core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -58,57 +60,60 @@ final case class InvokerInfo()
+    kv: KeyValueStore,
+    invokerDownCallback: String => Unit) extends Actor {
+
     implicit val transid = TransactionId.invokerHealth
     val logging = new AkkaLogging(context.system.log)
     implicit val timeout = Timeout(5.seconds)
     implicit val ec = context.dispatcher
 
-    /** Collects the status of all invokers */
-    def collectStatus = {
-        val list = context.children.map { invoker =>
-            val name = invoker.path.name
-            invoker.ask(GetStatus).mapTo[InvokerState].map(name -> _)
-        }
-        Future.sequence(list).map(_.toMap)
-    }
+    val invokers = mutable.HashMap.empty[String, ActorRef]
+    val invokerStatus = mutable.HashMap.empty[String, InvokerState]
 
     def receive = {
         case p: PingMessage =>
-            val invoker = context
-                .child(p.name) // get an existing actor
-                .getOrElse { // or create a new one lazily
-                    logging.info(this, s"registered a new invoker: ${p.name}")(TransactionId.invokerHealth)
-                    val ref = context.actorOf(InvokerActor.props, p.name)
-                    ref ! SubscribeTransitionCallBack(self) // register for state change events
-                    self ! StatusChange
-                    ref
-                }
+            val invoker = invokers.getOrElseUpdate(p.name, {
+                logging.info(this, s"registered a new invoker: ${p.name}")(TransactionId.invokerHealth)
+                val ref = childFactory(context, p.name)
+                ref ! SubscribeTransitionCallBack(self) // register for state change events
+                ref
+            })
             invoker.forward(p)
 
-        case GetStatus => pipe(collectStatus) to sender()
+        case GetStatus => sender() ! invokerStatus.toMap
 
-        case StatusChange => collectStatus.foreach { allState =>
-            val json = allState.mapValues(_.asString).toJson
-            kv.put(LoadBalancerKeys.invokerHealth, json.compactPrint)
+        case CurrentState(invoker, currentState: InvokerState) =>
+            invokerStatus.update(invoker.path.name, currentState)
+            publishStatus()
 
-            val pretty = allState.toSeq.sortBy {
-                case (name, _) => name.drop(7).toInt
-            }.map { case (name, state) => s"$name: $state" }
-            logging.info(this, s"invoker status changed to ${pretty.mkString(", ")}")
-        }
-
-        case Transition(invoker, oldState, newState) =>
-            self ! StatusChange
+        case Transition(invoker, oldState: InvokerState, newState: InvokerState) =>
+            invokerStatus.update(invoker.path.name, newState)
             newState match {
                 case Offline => Future(invokerDownCallback(invoker.path.name))
                 case _       =>
             }
+            publishStatus()
+    }
+
+    def publishStatus() = {
+        val json = invokerStatus.toMap.mapValues(_.asString).toJson
```

Comment:
`toMap.mapValues` is unnecessary, `.mapValues` skips building an immutable version of the map that is thrown away.